format-common/src/main/java/io/cdap/plugin/format/input/PathTrackingInputFormatProvider.java

```diff
@@ -63,6 +65,31 @@ protected void validate() {
     // no-op
   }
 
+  protected Schema addPathField(Schema dataSchema, FailureCollector failureCollector) {
+    return addPathField(failureCollector, dataSchema, conf.getPathField());
+  }
+
+  public static Schema addPathField(FailureCollector failureCollector, Schema dataSchema, String pathField) {
+    if (pathField == null) {
+      return dataSchema;
+    }
+    if (dataSchema == null) {
+      failureCollector.addFailure("Schema is null", "Please provide valid schema")
+        .withConfigProperty("nullSchema");
+      throw failureCollector.getOrThrowException();
+    }
+    List<Schema.Field> fields = new ArrayList<>(dataSchema.getFields());
+    for (Schema.Field field : fields) {
+      if (pathField.equals(field.getName())) {
+        failureCollector.addFailure("Pathfield already exists in the schema",
+          String.format("Please provide a new value instead of %s", pathField)).withConfigProperty("pathField");
```

Comment:
nit: Please provide a field that does not already exist.

It's better to put relevant error information earlier in the error message, rather than split across the error message and corrective action.

Reply:
Done